internal/controllers/utils/name_matching.go

```diff
@@ -4,11 +4,36 @@
 package utils
 
 import (
+	"reflect"
+
 	"sigs.k8s.io/controller-runtime/pkg/builder"
 	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
 	"sigs.k8s.io/controller-runtime/pkg/predicate"
+
+	capsulev1beta2 "github.com/projectcapsule/capsule/api/v1beta2"
 )
 
+var CapsuleConfigSpecChangedPredicate = predicate.Funcs{
+	// Only care about Updates – fire when Spec changed
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldObj, ok1 := e.ObjectOld.(*capsulev1beta2.CapsuleConfiguration)
+		newObj, ok2 := e.ObjectNew.(*capsulev1beta2.CapsuleConfiguration)
+		if !ok1 || !ok2 {
+			return false
+		}
+
+		return !reflect.DeepEqual(oldObj.Spec, newObj.Spec)
+		// or with k8s helpers:
+		// return !equality.Semantic.DeepEqual(oldObj.Spec, newObj.Spec)
```

Comment:
We should avoid at our best the usage of the `reflection` package since it's expensive.